smithy-model/src/main/java/software/amazon/smithy/model/loader/IdlModelParser.java

```diff
@@ -517,6 +524,12 @@ private void parseMember(
         String target = ParserUtils.parseShapeId(this);
 
         if (structureMember && peek() == '!') {
+            if (!modelFile.getVersion().supportsRequiredSugar()) {
+                throw syntax(memberId, String.format(
+                        "The '!' suffix can only be used on structure members when using Smithy 2.0 or later, but "
+                        + "you're using version `%s`. Make `$version: \"2\"` the first line of this file.",
```

Comment:
"Make \`$version: \"2\"\` the first line of this file to update" or something to indicate they're updating (pedantic i know but still)